src/DGtal/topology/CubicalComplex.h

```diff
@@ -0,0 +1,1307 @@
+    */
+    void fillData( Data data = Data() );
+
+    /**
+    * Fills the data of every cell of dimension \a d this cubical
+    * complex, which becomes \a data. Default value resets the data to
+    * zero.
+    *
+    * @param d the dimension of cell \a aCell.
+    * @param data any data.
+    */
+    void fillData( Dimension d, Data data = Data() );
+
+    /**
+    * @return the maximal dimension of a cell in the complex, 0 if
+    * the complex is empty.
+    */
+    Dimension dim() const;
+
+    /**
+    * @param aCell any cell valid in the Khalimsky space associated to the complex.
+    * @return the dimension of the cell \a aCell.
+    */
+    Dimension dim( const Cell& aCell ) const;
+
+    /**
+    * @param d the dimension of cells.
+    * @return the number of cells of dimension \a d in this complex.
+    */
+    Size nbCells( Dimension d ) const;
+
+    /**
+    * @note For instance, all Platonician solids have euler number
+    * equal to one, while their surface have euler number equal to
+    * two.
+    *
+    * @return the Euler number of this complex which equals nbCells( 0 ) - nbCells( 1 ) + nbCells( 2 ) - ...
+    */
+    Integer euler() const;
+
+    /**
+    * @return a reference to the Khalimsky space associated to this complex.
+    */
+    const KSpace& space() const;
+
+    // ---------- cell container operations ---------------
+  public:
+
+    /// @return an iterator pointing on the first cell of this complex
+    /// (lower dimensional cells come first).
+    ConstIterator begin() const;
+
+    /// @return an iterator pointing after the last cell of this complex
+    /// (upper dimensional cells arrive last).
+    ConstIterator end() const;
+
+    /// @return an iterator pointing on the first cell of this complex
+    /// (lower dimensional cells come first).
+    Iterator begin();
+
+    /// @return an iterator pointing after the last cell of this complex
+    /// (upper dimensional cells arrive last).
+    Iterator end();
+
+    /**
+     * @param aCell any cell.
+     * @return the number of matches for \a aCell, which is thus zero (not present) or one (present).
+     */
+    Size count( const Cell& aCell ) const;
+
+    /// @return the total number of cells in this complex.
+    Size size() const;
+
+    /// @return the maximal number of cells in this complex (i.e., the
+    /// number of cells of the Khalimsky space).
+    Size max_size() const;
+
+    /// @return 'true' if and only if the complex does not hold any cell.
+    bool empty() const;
+    
+    /**
+     * Get range of equal elements to \a aCell. Because all elements
+     * in a set container are unique, the range returned will contain
+     * a single element at most.
+     *
+     * @param aCell any cell.
+     *
+     * @return the bounds of a range that includes all the elements in
+     * the container that are equivalent to val.
+     */
+    std::pair< ConstIterator, ConstIterator > equal_range( const Cell& aCell ) const;
```

Comment:
This is to satisfy STL concepts for associative containers, so I think it is better to keep them lower case.